### ISSUE TITLE
Check that promise is not fulfilled too early

### ISSUE THREAD

**Issue by bclozel** [cujojs/cram#41]
When using cram as a JavaScript API, it seems that cram is fulfilling its promise _before_ the actual output file is written/flushed to disk.

``` js
var promise = when(cram(opts));
var onFulfill = function() {
  if(!fs.existsSync(outputPath)) {
    // this exception is thrown
    throw "missing cram output file: "+outputPath;
  }
  //...
}
promise.done(onFulfill, cramError);
```

**Comment by unscriptable** [comment]
Hey @bclozel, this is working in the dev branch.  I did some functional-level testing and it seems to work correctly now.  Feel free to try it out. -- J

**Comment by bclozel** [comment]
It works! Great!
I'll release my plugin on github with an ugly `"cram": "https://github.com/cujojs/cram/tarball/dev"` dependency declaration and fix that before pushing a real version on npmjs.

**Comment by briancavalier** [comment]
Nice work, guys!

**Comment by unscriptable** [comment]
cool!  I've got another issue to fix in cram, so i'll try to release a 0.8.1 later this week.  -- J
